### PR TITLE
Add CausalLM Qwen3-0.6B verification test and X86 CI

### PR DESCRIPTION
**Pull request by EunjuYang**
## Summary
- Add end-to-end verification test for CausalLM that compares C++ inference output against Python (HuggingFace) reference using Qwen3-0.6B model
- Python script downloads model, converts weights, runs greedy inference, and saves reference token IDs + prefill logits
- C++ GTest verifies both prefill logits (top-K with 1e-3 tolerance) and generated token IDs match exactly
- New GitHub Actions X86 CI workflow on Ubuntu 24.04, triggered on PRs (CausalLM path changes) and manual dispatch

## New Files
| File | Description |
|------|-------------|
| `Applications/CausalLM/test/generate_reference.py` | Downloads Qwen3-0.6B, converts weights, generates reference outputs |
| `Applications/CausalLM/test/test_causallm_qwen3.cpp` | GTest verifying prefill logits and generated tokens vs Python |
| `Applications/CausalLM/test/meson.build` | Build config for the test binary |
| `Applications/CausalLM/res/qwen3/qwen3-0.6b/nntr_config.json` | Qwen3-0.6B model config for nntrainer |
| `Applications/CausalLM/res/qwen3/qwen3-0.6b/weight_converter.py` | Weight converter for Qwen3-0.6B |
| `.github/workflows/causallm_x86_test.yml` | X86 CI workflow |

## Test plan
- [ ] CI workflow runs successfully on Ubuntu 24.04 x86_64
- [ ] Python reference generator downloads Qwen3-0.6B and produces reference data
- [ ] C++ test binary loads converted weights and runs inference
- [ ] Prefill logits top-K values match within 1e-3 tolerance
- [ ] All 10 greedy-decoded token IDs match between C++ and Python

https://claude.ai/code/session_01XV3MuDVqKYJxNVr1eoAMw3

### CODE REVIEW

File: Applications/CausalLM/res/qwen3/qwen3-0.6b/weight_converter.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## @file weight_converter.py
+## @brief weight conversion script for qwen3 0.6b model
+## @author Eunju Yang <ej.yang@samsung.com>
+
+import argparse
+import torch
+import numpy as np
+from transformers import AutoConfig, AutoTokenizer, AutoModelForCausalLM
+
+total_size = 0
```
Comment (EunjuYang) [inline Applications/CausalLM/res/qwen3/qwen3-0.6b/weight_converter.py]: **Cleanup: 사용되지 않는 변수들**

- `total_size` (line 10): 선언만 되고 사용되지 않음
- `device` (line 73): 계산만 하고 사용되지 않음 (모델은 기본 CPU에 로드)
- `tokenizer` (line 71): 로드만 하고 사용되지 않음

불필요한 코드를 제거하면 좋겠습니다.

**Review by EunjuYang — COMMENTED** [review]
## 전체 리뷰 요약

Qwen3-0.6B CausalLM 검증 테스트와 X86 CI 추가 PR입니다. 전체적인 구조와 접근 방식이 잘 설계되어 있습니다. Python reference → C++ GTest 비교 패턴이 깔끔하고, `causal_lm.cpp`/`transformer.cpp`의 JSON 파싱 방어 코드도 적절합니다.

### 수정이 필요한 사항 (Bugs)
1. **NPY 파서 v2.0 미지원** - `load_npy()`가 header_len을 2바이트로만 읽어서, 큰 배열에서 깨질 수 있음
2. **메모리 누수** - `testInference()`에서 `input_sample`이 예외 시 해제 안 됨 → `std::vector` 사용 권장
3. **CI: `g++` alternatives 설정 누락** - `gcc`만 설정하고 `g++`은 `--set` 안 함
4. **`setup-env.sh`: `sudo` 없는 `apt-get`** - non-root 환경에서 실패

### 개선 권장 사항
5. **`ROPE_THETA` 미초기화 경고 없음** - config에 값이 없으면 조용히 넘어감
6. **`generate_reference.py`의 KV-cache 미사용** - 매 스텝 전체 재계산
7. **HF 캐시 키에 버전 정보 없음** - 모델 업데이트 시 stale cache 가능

### Cleanup
8. 미사용 import/변수: `shutil` (generate_reference.py), `total_size`/`device`/`tokenizer` (weight_converter.py)
9. Copyright 연도 불일치 (2025 vs 2026)

1~4번은 수정을 권장합니다. 나머지는 선택사항입니다.
